### ISSUE TITLE
Deprecation warning doesn't show zero in version of 1.10 at the end

### ISSUE THREAD

**Issue by iionly** [Elgg/Elgg#7841]
> WARNING: Deprecated in 1.1: The view output/confirmlink has been deprecated

Second parameter of the corresponding elgg_deprecated_notice() is not a string in output/confirmlink
```php
elgg_deprecated_notice('The view output/confirmlink has been deprecated, please use output/url', 1.10);
```
Question is:

* require the second paramater to be a string (= fix in call in output/confirmlink),
* or fix in elgg_deprecated_notice() (if that's even possible and the zero is not already lost at this point)

Next version where this problem will occur is likely 2.0 (or will there be no deprecation notices included there anyway?).

**Comment by beck24** [comment]
I thought this had been fixed already...

**Comment by Srokap** [comment]
I think we should just force string params and be done with it. Versions in general are not floating point numbers, so shouldn't be treated as such. I also remember very similar bug due to versions being treated as numbers, but I think it was internal processing of version comparison.

**Comment by mrclay** [comment]
@Srokap agreed, as in convert all uses to string and throw if not given a string.

**Comment by juho-jaakkola** [comment]
Fixed in #7849
